docs/adrs/0077-eject-cra-app.md

```diff
@@ -0,0 +1,108 @@
+---
+title: '0077 Eject the application from Create React App'
+description: |
+  Decision outcome: To be decided
+---
+
+# Eject the Create React App configuration
+
+<!-- **User Story:** *[ticket/issue-number]* -->
+
+## Background
+
+In the previous [ADR 0072 Using React-App-Rewired][adr-0072], the decision of
+ejecting the React application was proposed as an alternative to leveraging
+_React-App-Rewired_. _React-App-Rewired_ has been lightly maintained since
+_Create-React-App_ version 2.0. It is also stated in the _React-App-Rewired_
+repository that with the use of the library the configurations are now owned by
+the team that leverages the tool. In other words, the MilMove React application
+has technically been ejected since ADR 0072 has been implemented in the
+repository.
+
+[adr-0072]: ./0072-using-react-app-rewired.md
+
+The main reason to consider this again is that it was discovered on back on the
+13th of May 2022 that the _React-Scripts_, which _Create-React-App_ provides,
+does not call `compiler.close` which prevents the *webpack* cache from being
+saved. This means that even without changes being made to the `src/` directory
+the `make client_build` command will always compile the application every time.
+This prevents the CI system from saving artifacts from previous builds and means
+that the CI system compiles the client code on every time regardless of changes
+made to that single component of the MilMove system.
+
+As pointed out in DP3 Slack 🔒, there's a significant speed up in Client
+compilation times in CI when the *webpack* cache is enabled. According to
+*webpack*'s own documentation on this, the file system cache should be enabled
+for CI/CD systems.
+
+[🔒 DP3 Slack, discussing compile times of ~15 seconds when there are no Client app changes.](https://ustcdp3.slack.com/archives/CTQQJD3G8/p1672775265604429)
+
+[➡️  *webpack*: Setup cache in CI/CD system](https://webpack.js.org/configuration/cache/#setup-cache-in-cicd-system)
+
+There is also evidence from the _Create-React-App_ repository that is has been
+volunteer-supported since the 6th of July 2021 according to one of the original
+co-authors of the library. This means that the future of _Create-React-App_ is
+unknown for our project. Since, _Create-React-App_ is only a set of useful
```

Comment:
```suggestion
According to one of the original co-authors of the library, the _Create-React-App_ 
repository has been volunteer-supported since the 6th of July 2021. This means that 
the future of _Create-React-App_ is unknown for our project. 
Since _Create-React-App_ is only a set of useful
```